server/src/controllers/auth.controller.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import '@babel/polyfill';
+import models from '../database/models';
+import { encrpytPassword, generateToken, comparePassword } from '../helper/encryption';
+import getSignupUserData from '../utils/user.utils';
+
+
+const { Users } = models;
+
+const signupUser = async (req, res) => {
+  const userData = getSignupUserData(req.body);
+
+  const existingUser = await Users.findOne({
+    where: {
+      email: userData.email
+    }
+  });
+
+  if (existingUser) {
+    return res.status(409).json({ error: 'User with that email already exists' });
+  }
+
+  const hashPassword = await encrpytPassword(userData.password);
+
+  const newUser = await Users.create({ ...userData, password: hashPassword });
+
+  const payload = {
+    newUser
+  };
+
+  const token = generateToken(payload);
+
+  newUser.password = undefined;
+
+  const data = { ...newUser.get(), token };
+
+  return res.status(201).json({ status: 201, data });
```

Comment:
I'm not sure we are adding any value returning status code in the response body and also setting the same status as the HTTP header. Rather, the request body can have  `status: 'success'` or 'failure'. You can also decide to use another convention `success: true` or false.

Reply:
ok